### PR TITLE
less repeatin g code for logpdf / entropy / kldivergence

### PR DESCRIPTION
**Pull request by filtron**

### CODE REVIEW

File: src/normal/normal_generic.jl [hunk 1]
```diff
@@ -60,7 +60,8 @@ entropy(N::AbstractNormal) = entropy(N)
 
 Computes the Kullback-Leibler divergence between N1 and N2.
 """
-kldivergence(N1::AbstractNormal, N2::AbstractNormal) = kldivergence(N1, N2)
+kldivergence(N1::AbstractNormal{T}, N2::AbstractNormal{T}) where {T<:Number} =
```
Comment (nathanaelbosch) [inline src/normal/normal_generic.jl:63]: Are there even `AbstractNormal{T}`s where `T` is not a `Number`? If not, you can simplify it again imo.

Comment (nathanaelbosch) [reply src/normal/normal_generic.jl:63]: So according to https://github.com/filtron/MarkovKernels.jl/blob/70c29bd75e250dbb079bba95e63b7819da0e5384/src/normal/normal.jl#L6 it's always a number, so no need to specialize?

File: src/normal/normal.jl [hunk 2]
```diff
@@ -32,15 +32,26 @@ std(N::Normal) = sqrt.(var(N))
...
+_nscale(T::Type{<:Complex}) = one(real(T))
 
-entropy(N::Normal{T,U,V}) where {T<:Real,U,V} =
-    dim(N) / 2.0 * (log(2.0 * π) + 1) + logdet(N.Σ) / 2.0
-entropy(N::Normal{T,U,V}) where {T<:Complex,U,V} = dim(N) * (log(π) + 1) + logdet(N.Σ)
+_piconst(T::Type{<:Real}) = T(2π)
+_piconst(T::Type{<:Complex}) = real(T)(π)
 
+logpdf(N::Normal, x) =
+    -_nscale(eltype(N)) * (logdet(_piconst(eltype(N)) * N.Σ) + norm_sqr(residual(N, x)))
```
Comment (nathanaelbosch) [inline src/normal/normal.jl:42]:
```suggestion
logpdf(N::Normal{T}, x) where {T} =
    -_nscale(T) * (logdet(_piconst(T) * N.Σ) + norm_sqr(residual(N, x)))
```
JuliaFormatter might complain though so you'd need to fix that afterwards.

File: src/normal/normal.jl [hunk 3]
```diff
@@ -32,15 +32,26 @@ std(N::Normal) = sqrt.(var(N))
...
+entropy(N::Normal) =
+    _nscale(eltype(N)) * (dim(N) * (log(_piconst(eltype(N))) + 1) + logdet(N.Σ))
```
Comment (nathanaelbosch) [inline src/normal/normal.jl:44]: Same as above:
```suggestion
entropy(N::Normal{T}) where {T} =
    _nscale(T) * (dim(N) * (log(_piconst(T) + 1) + logdet(N.Σ))
```

**Review by nathanaelbosch — CHANGES_REQUESTED** [review]

**Review by filtron — COMMENTED** [review]
seems fine.

**Review by nathanaelbosch — APPROVED** [review]

**Pull request merged**